pkg/resource/v1/certconfig/current.go

```diff
@@ -2,8 +2,67 @@ package certconfig
 
 import (
 	"context"
+	"fmt"
+
+	"github.com/giantswarm/apiextensions/pkg/apis/core/v1alpha1"
+	"github.com/giantswarm/cluster-operator/pkg/resource/v1/certconfig/key"
+	"github.com/giantswarm/microerror"
+	"github.com/giantswarm/randomkeytpr"
+	"k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 )
 
+// GetCurrentState takes observed custom object as an input and based on that
+// information looks for current state of cluster certconfigs and returns them.
+// Return value is of type []*v1alpha1.CertConfig.
 func (r *Resource) GetCurrentState(ctx context.Context, obj interface{}) (interface{}, error) {
-	return nil, nil
+	clusterGuestConfig, err := r.toClusterGuestConfigFunc(obj)
+	if err != nil {
+		return nil, microerror.Mask(err)
+	}
+
+	r.logger.LogCtx(ctx, "debug", "looking for a list of certconfigs in the Kubernetes API")
+
+	var certConfigs []*v1alpha1.CertConfig
+	{
+		labelSelector := &metav1.LabelSelector{}
+		labelSelector = metav1.AddLabelToSelector(labelSelector, randomkeytpr.ClusterIDLabel, key.ClusterID(*clusterGuestConfig))
+
+		selector, err := metav1.LabelSelectorAsSelector(labelSelector)
+		if err != nil {
+			return nil, microerror.Mask(err)
+		}
+
+		listOptions := metav1.ListOptions{
+			LabelSelector: selector.String(),
+			Limit:         listCertConfigLimit,
+		}
+
+		continueToken := ""
+
+		for {
+			listOptions.Continue = continueToken
+
+			certConfigList, err := r.g8sClient.CoreV1alpha1().CertConfigs(v1.NamespaceDefault).List(listOptions)
+			if err != nil {
+				return nil, microerror.Mask(err)
+			}
+
+			for _, item := range certConfigList.Items {
+				// Make a copy of an Item in order to not refer to loop
+				// iterator variable.
+				item := item
+				certConfigs = append(certConfigs, &item)
+			}
+
+			continueToken = certConfigList.Continue
```

Comment:
For my understanding, is this some kind of paging the k8s client listing implements?

Reply:
Yes. See `Limit` field in [ListOptions](https://godoc.org/k8s.io/apimachinery/pkg/apis/meta/v1#ListOptions).